server/server.go

```diff
@@ -2524,8 +2524,10 @@ func (s *Server) Shutdown() {
 	if s.websocket.server != nil {
 		doneExpected++
 		s.websocket.server.Close()
```

Comment:
Should access to websocket.server also be protected?

Reply:
We don't need because we set it to `nil` only below. The locking below is because we set the server/listener to `nil` while some code elsewhere is accessing some websocket.server/listener functions under the RLock().

But I did add same locking in lameDuckMode() where we are also setting those two to nil.

Reply:
ok but if we did parallel shutdowns it could potentially race technically.. I agree probably not needed here to capture under lock for test of nil.